proto/substrait/algebra.proto

```diff
@@ -52,6 +53,22 @@ message RelCommon {
       substrait.extensions.AdvancedExtension advanced_extension = 10;
     }
   }
+
+  // Optional metadata that the producer may attach to the relation.
+  // Consumer should hold no expectations about the existance metadata.
```

Comment:
How about:  "The consumer should not rely on the metadata for the evaluation of the plan."

Reply:
Might even want to mention that both the consumer and the producer shouldn't rely on this.

In other words:

 * As a consumer, that wants to handle plans from multiple producers, be prepared to handle plans that are missing this metadata
 * As a producer, that wants plans to be run by multiple consumers, understand that your consumer might ignore this metadata